### PR TITLE
Add sort for deadline

### PR DESCRIPTION
**Pull request by zsh-eng**
Currently, we cannot sort by internship deadline.

Let's add sorting by internship deadline to the existing sort logic.

### CODE REVIEW

**Review by tim-pipi — COMMENTED** [review]
LGTM! Just one thing to confirm, does this sort by alphabetical order, or does it sort by the actual dates? Because currently, deadline is inputted as a String and then parsed as a Date (to check for validity) and then is stored back as a String.

**Review by AprupKale — COMMENTED** [review]
LGTM 👍!

File: src/main/java/seedu/letsgethired/model/application/Deadline.java
Comment (tim-pipi) [inline src/main/java/seedu/letsgethired/model/application/Deadline.java:1]: Nice! Thanks for the changes! This should now work as intended!

File: src/main/java/seedu/letsgethired/model/application/InternApplicationComparator.java
Comment (tim-pipi) [inline src/main/java/seedu/letsgethired/model/application/InternApplicationComparator.java:1]: Nice! Now the application is more scalable and extensible!

**Review by tim-pipi — APPROVED** [review]
LGTM! Thanks for the changes!

**Review by AprupKale — APPROVED** [review]
LGTM 👍! Thanks for the changes @zsh-eng

**Pull request merged**